navigation/navigation-runtime-ktx/src/androidTest/java/androidx/navigation/NavControllerTest.kt

```diff
@@ -41,6 +48,29 @@ class NavControllerTest {
             DESTINATION_ID in graph
         )
     }
+
+    @Test
+    @Suppress("EXPERIMENTAL_API_USAGE")
+    fun getBackStackEntryFlow() = runBlocking {
+        navController.graph = navController.createGraph(startDestination = 1) {
+            test(1)
+            test(2)
+            test(3)
+        }
+
+        navController.getBackStackEntryFlow()
+            .take(navController.graph.count())
+            .withIndex()
+            .onEach { (index, backStackEntry) ->
+                val expectedDestination = index + 1
+                assertEquals(expectedDestination, backStackEntry.destination.id)
```

Comment:
Please use [Truth](https://truth.dev/) asserts (note, we already have an androidTest dependency on Truth) over Junit asserts here:
```
assertThat(backStackEntry.destination.idt).isEqualTo(expectedDestination)
```
Or preferably use the `assertWithMessage()` version that includes a more detailed message that is printed when the assert fails.